File: C++/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    
    map<string, int> mp;
    
    for(string pa : participant){
        mp[pa] += 1;
    }
    
    for(string com : completion){
        mp[com] -= 1;
    }
    
    for(auto &p : mp){
        if(p.second > 0){
            return p.first;
        }
    }
    
    return "";
}

File: Python3/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    answer = ''
    di = defaultdict(int)
    
    for p in participant:
        di[p] += 1
    for c in completion:
        di[c] -= 1
    
    keys = list(di.keys())
    
    for key in keys:
        if(di[key] != 0):
            return key
    
    return answer